GitUI/CommandsDialogs/FormSubmodules.cs

```diff
@@ -119,7 +120,12 @@ private void SynchronizeSubmoduleClick(object sender, EventArgs e)
         {
             using (WaitCursorScope.Enter())
             {
-                FormProcess.ShowDialog(this, GitCommandHelpers.SubmoduleSyncCmd(SubModuleLocalPath.Text));
+                bool success = FormProcess.ShowDialog(this, process: null, arguments: GitCommandHelpers.SubmoduleSyncCmd(SubModuleLocalPath.Text), Module.WorkingDir, input: null, useDialogSettings: true);
+                if (!success)
+                {
+                    return;
+                }
+
                 Initialize();
```

Comment:
`Initialize` must be safe on error, too. Git could have already done things before it fails.

Reply:
Same here, if `submodules sync` operation failed is there a point to initialise anything?